zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/zeebe/ExecutionListenerTransformer.java

```diff
@@ -12,28 +12,71 @@
 import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeExecutionListener;
 import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeExecutionListenerEventType;
 import io.camunda.zeebe.protocol.record.value.ExecutionListenerEventType;
+import io.camunda.zeebe.util.Either;
+import io.camunda.zeebe.util.buffer.BufferUtil;
 import java.util.Collection;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public final class ExecutionListenerTransformer {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(ExecutionListenerTransformer.class);
+
   public void transform(
       final ExecutableFlowNode executableFlowNode,
       final Collection<ZeebeExecutionListener> executionListeners,
       final ExpressionLanguage expressionLanguage) {
 
     executionListeners.forEach(
-        listener -> addListenerToFlowNode(listener, executableFlowNode, expressionLanguage));
+        listener -> {
+          try {
+            addListenerToFlowNode(listener, executableFlowNode, expressionLanguage);
+          } catch (final Exception e) {
+            // This is an additional safety, as our transformers currently require additional
+            // safeguards to avoid breaking the partition on replay
+            LOGGER.warn(
+                """
+                Failed to transform execution listener for flow node '{}', \
+                caught an unexpected exception: '{}': {}. Ignoring this listener definition.""",
+                BufferUtil.bufferAsString(executableFlowNode.getId()),
+                e.getClass().getSimpleName(),
+                e.getMessage());
+          }
+        });
   }
 
   private void addListenerToFlowNode(
       final ZeebeExecutionListener listener,
       final ExecutableFlowNode flowNode,
       final ExpressionLanguage expressionLanguage) {
+    Either.<Error, ZeebeExecutionListener>right(listener)
+        .flatMap(l -> requireNotNull(l, () -> l, "listener"))
+        .flatMap(l -> requireNotNull(l, l::getEventType, "eventType"))
+        .flatMap(l -> requireNotNull(l, l::getType, "type"))
+        .flatMap(l -> requireNotNull(l, l::getRetries, "retries"))
+        .ifRightOrLeft(
+            ok ->
+                flowNode.addListener(
+                    fromZeebeExecutionListenerEventType(listener.getEventType()),
```

Comment:
The usage of this method was a conflict. I've simply let IntelliJ combine the two solutions.